### PR TITLE
[ENG-976][document-picker] convert to kotlin

### PR DESCRIPTION
**Pull request by dsokal**
# Why

https://linear.app/expo/issue/ENG-976/convert-expo-document-picker-to-kotlin

This is part of the effort to convert our Android codebase to Kotlin. This PR converts `expo-document-picker` to Kotlin.

# How

- I used the auto-converter from Android Studio (`Code > Covert Java File to Kotlin File`).
- Then, I went over the code and refactored it here and there. 
- I handled more errors: 
  - When `getDocumentAsync` receives invalid `options`.
  - When the module fails to read the selected document.
  - When the module fails to copy the document to the cache directory.

# Test Plan

I tested the module with NCL (in `bare-expo`). I haven't seen any regression.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        if (!mCopyToCacheDirectory || originalDocumentDetails == null) {
+          originalDocumentDetails
+        } else {
+          val copyPath = copyDocumentToCacheDirectory(uri, originalDocumentDetails.name)
+          if (copyPath == null) {
+            mPromise!!.reject("E_DOCUMENT_PICKER", "Failed to copy to cache directory.")
+            mPromise = null
+            return
+          } else {
+            originalDocumentDetails.copy(uri=copyPath)
```
Comment (lukmccall) [inline packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt:84]:
```suggestion
            originalDocumentDetails.copy(uri = copyPath)
```

File: packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      return null
+    } finally {
+      outputStream?.also {
+        try {
+          it.close()
+        } catch (e: IOException) {
+          e.printStackTrace()
+        }
+      }
+    }
```
Comment (lukmccall) [inline packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt:138]:
```suggestion
	try {
	  context.contentResolver.openInputStream(documentUri).use { inputStream -> 
	    FileOutputStream(File(outputFilePath)).use { outputStream -> 
	  	  IOUtils.copy(inputStream, outputStream)
	  	}
	  }
	} catch (e: IOException) {
      e.printStackTrace()
      return null
	}
```

**Review by lukmccall — CHANGES_REQUESTED** [review]
LGTM ✅
However, I've left some suggestions ;) 

Also, I think it will be better in `onActivityResult` to get rid of the nullable promise type earlier. After first if, you can create a local variable with not nullable type:
```kt
val promise = mPromise!!
mPromise = null
```
In that way, you don't have to unwrap a promise whenever you want to use it.

**Review by lukmccall — APPROVED** [review]
🚀🚀🚀

**Review by bbarthec — APPROVED** [review]
Solid wo, just this one inverted condition 😉

**Pull request merged**